roles/st2web/tasks/main.yml

```diff
@@ -1,94 +1,5 @@
 ---
 # tasks file for st2web
-
-- name: Add nginx key
-  become: yes
-  apt_key:
-    url: http://nginx.org/keys/nginx_signing.key
-    state: present
-
-- name: Add nginx repos
-  become: yes
-  apt_repository:
-    repo: "deb http://nginx.org/packages/ubuntu/ trusty nginx"
-    state: present
-    update_cache: yes
-
-- name: Install nginx
-  become: yes
-  apt:
-    name: nginx=1.8.*
-    state: present
-
-- name: Install latest st2web package
-  become: yes
-  apt:
-    name: st2web
-    state: latest
-  when: st2_version == "stable"
-  notify:
-    - restart st2
-
-- name: Install exact st2web package
-  become: yes
-  apt:
-    name: st2web={{ st2_version }}-{{ st2web_revision }}
-    state: present
-  when: st2_version != "stable"
-  notify:
-    - restart st2
-
-- name: Create ssl cert dir
-  become: yes
-  file:
-    state: directory
-    dest: /etc/ssl/st2
-
-- name: Create ssl cert
-  become: yes
-  shell: openssl req -x509 -newkey rsa:2048 -keyout /etc/ssl/st2/st2.key -out /etc/ssl/st2/st2.crt -days 365 -nodes -subj "/C=US/ST=California/L=Palo Alto/O=StackStorm/OU=Information Technology/CN=$(hostname)"
-  args:
-    creates: /etc/ssl/st2/st2.key
-  notify:
-    - Restart nginx
-
-- name: Remove default site
-  become: yes
-  file:
-    path: /etc/nginx/sites-enabled/default
-    state: absent
-
-- name: Create nginx folders
-  become: yes
-  file:
-    state: directory
-    path: "{{ item }}"
-  with_items:
-    - /etc/nginx/sites-available/
-    - /etc/nginx/sites-enabled/
-
-- name: Ensure site-available is loaded
-  become: yes
-  lineinfile:
-    state: present
-    dest: /etc/nginx/nginx.conf
-    regexp: 'include /etc/nginx/sites-enabled/'
-    insertafter: '    include /etc/nginx/conf.d/'
-    line: 'include /etc/nginx/sites-enabled/*;'
-    
-- name: Copy Nginx config
-  become: yes
-  command: cp /usr/share/doc/st2/conf/nginx/st2.conf /etc/nginx/sites-available/
-  args:
-    creates: /etc/nginx/sites-available/st2.conf
-  notify: 
-    - Restart nginx
-
-- name: Enable site
-  become: yes
-  file:
-    state: link
-    src: /etc/nginx/sites-available/st2.conf
-    dest: /etc/nginx/sites-enabled/st2.conf
-  notify:
-    - Restart nginx
+- name: Install st2web on {{ ansible_distribution }}
+  include: st2web_{{ ansible_pkg_mgr }}.yml
```

Comment:
Looks like a lot of code duplication here.
No sense for splitting, 2 files are differing with just a couple of small things at the moment.

Your previous approach with `package` module would be more elegant, see: https://github.com/StackStorm/ansible-st2/blob/32566695905cb92a4a975eba01e564702554b2fd/roles/st2mistral/tasks/main.yml#L12